### PR TITLE
WithLicense functionality added.

### PR DESCRIPTION
**Pull request by DuncanBeutler**

### CODE REVIEW

File: examples/international_street.js [hunk 1]
```diff
@@ -9,9 +9,10 @@ const Lookup = SmartyStreetsSDK.internationalStreet.Lookup;
 
 // for client-side requests (browser/mobile), use this code:
 let key = process.env.SMARTY_WEBSITE_KEY;
-const  credentials = new SmartyStreetsCore.SharedCredentials(key);
+const credentials = new SmartyStreetsCore.SharedCredentials(key);
 
 let client = SmartyStreetsCore.buildClient.internationalStreet(credentials);
+// .withLicense("us-street-cloud");
```
Comment (mdwhatcott) [inline examples/international_street.js:15]: Let's go with `"us-core-cloud"` here.

File: examples/us_street.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ const Lookup = SmartyStreetsSDK.usStreet.Lookup;
 let key = process.env.SMARTY_WEBSITE_KEY;
 const  credentials = new SmartyStreetsCore.SharedCredentials(key);
 let client = SmartyStreetsCore.buildClient.usStreet(credentials);
+	// .withLicense("us-street-cloud");
```
Comment (mdwhatcott) [inline examples/us_street.js:14]: Let's go with `"us-core-cloud"` here.

File: src/ClientBuilder.js [hunk 3]
```diff
@@ -129,6 +131,12 @@ class ClientBuilder {
 		return this;
 	}
 
+	withLicense(license) {
```
Comment (mdwhatcott) [inline src/ClientBuilder.js:134]: This option should be plural: 
```
withLicenses(licenses) {
   this.licenses = licenses;
   return this;
}
```

Comment (mdwhatcott) [reply src/ClientBuilder.js:134]: (and commented calls to this function should be updated accordingly)

Comment (DuncanBeutler) [reply src/ClientBuilder.js:134]: We had made the decision in the Go SDK to have the method receive a single license and have the option to be called multiple times. Do we want this to be consistent across all SDKs?

File: tests/test_LicenseSender.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const chai = require("chai");
```
Comment (camiblanch) [inline tests/test_LicenseSender.js:1]: Use tabs instead of spaces in this file

File: tests/test_LicenseSender.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	let request;
+	let licenses;
+	let licenseSender;
+
+	beforeEach(() => {
+		innerSender = {
+			send: () => true
+		};
+		request = new Request();
+
```
Comment (camiblanch) [inline tests/test_LicenseSender.js:18]: Remove this extra new line
```suggestion
```

File: tests/test_LicenseSender.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+		licenseSender.send(request);
+
+		expect(request.parameters).contains({"license":"0,1,2"});
+	});
+
+	it("doesn't append license to query if array is empty.", function () {
+		licenses = [];
+		licenseSender = new LicenseSender(innerSender, licenses);
+		licenseSender.send(request);
+		expect(request.parameters).to.not.have.property("license");
```
Comment (camiblanch) [inline tests/test_LicenseSender.js:33]: Add an empty line here to be consistent across the test syntax

```suggestion
		
		expect(request.parameters).to.not.have.property("license");
```

File: src/LicenseSender.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+const Promise = require("promise");
```
[the remark this answers is not in the document]
Comment (DuncanBeutler) [reply src/LicenseSender.js:1]: In the intelliJ IDE, everything is showing up as tabs - I'm not sure why that is...
Which IDE are you using?

**Pull request merged**